docs/patched_pdoc.py

```diff
@@ -20,93 +19,86 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
+"""A script that patches some pdoc functionality before calling it."""
+import datetime
+import os
+import pathlib
+import sys
 
-import json
-import os.path as path
-import re
-
-import pdoc
-from pdoc import cli
-
-
-def _patched_generate_lunr_search(modules, index_docstrings, template_config):
-    # This will only be called once due to how we generate the documentation, so we can ignore the rest
-    assert len(modules) == 1, "expected only 1 module to be generated, got more"
-    top_module = modules[0]
-
-    def trim_docstring(docstring):
-        return re.sub(
-            r"""
-            \s+[-=~]{3,}\s+|       # title underlines
-            ^[ \t]*[`~]{3,}\w*$|   # code blocks
-            \s*[`#*]+\s*|          # common markdown chars
-            \s*([^\w\d_>])\1\s*|   # sequences of punct of the same kind
-            \s*</?\w*[^>]*>\s*     # simple HTML tags
-            \s+                    # whitespace sequences
-        """,
-            " ",
-            docstring,
-            flags=re.VERBOSE | re.MULTILINE,
+import minify_html
+import sphobjinv
+from pdoc import __main__ as pdoc_main
+from pdoc import doc as pdoc_doc
+from pdoc import render as pdoc_render
+from pdoc.render import env as pdoc_env
+
+sys.path.append(os.getcwd())
+
+import hikari
+
+pdoc_html_module = pdoc_render.html_module
+
+
+def patched_html_module(*args, **kwargs) -> str:
+    # We patch this to minify the HTML output before sending it further
+    output = pdoc_html_module(*args, **kwargs)
+
+    return minify_html.minify(output, minify_js=True, minify_css=True)
+
+
+pdoc_render.html_module = patched_html_module
+
+# '-o' is the flag to provide the output dir. If it wasn't provided, we don't output the inventory
+generate_inventory = "-o" in sys.argv
+if generate_inventory:
+    project_inventory = sphobjinv.Inventory()
+    project_inventory.project = "hikari"
+    project_inventory.version = hikari.__version__
+
+    type_to_role = {
+        "module": "module",
+        "class": "class",
+        "function": "func",
+        "variable": "var",
+    }
```

Comment:
This produce a bug, when you trying call `:func:some_func` this will not find anything, because all of your functions named wrong.

From my debug session in `sphinx.ext.intersphinx` I understood that intersphinx resolve `:func:` to `:function:` and search in `objects.inv` only `:function:` role. So I can't just reference to any function from your docs.

As temp fix I could patch your `objects.inv` with `sphobjinv`, but this really weird.

Reply:
Create patch, and it working:
```py
def get_patched_hikari_inv() -> str:
    inv = sphobjinv.Inventory(url="https://davfsa.github.io/hikari-docs/stable/objects.inv")
    for object_ in inv.objects:
        if object_.role == "func" or object_.role == "var":
            object_.role = {"func": "function", "var": "variable"}[object_.role]
    sphobjinv.writebytes("./_build/hikari_objects.inv", sphobjinv.compress(inv.data_file()))
    return "./_build/hikari_objects.inv"

intersphinx_mapping = {
    "hikari": ("https://davfsa.github.io/hikari-docs/stable/", get_patched_hikari_inv()),
}
```

Reply:
After looking at the defined syntax, you are right. I was under the impression it should have been `func` and `var` because it is what were using before (https://github.com/hikari-py/hikari/blob/master/docs/documentation.mako#L550-L557, https://github.com/hikari-py/hikari/blob/master/docs/documentation.mako#L602-L610). These lines could be removed and then it will set it to `variable` and `function` accordingly

ref: https://sphobjinv.readthedocs.io/en/latest/syntax.html

Reply:
You had `objects.inv` in docs v1.... Could you push fix for this in master, because this not concern with this PR?

Reply:
Nvm, I create PR #1173.